src/child_process_supervisor.rs

```diff
@@ -108,13 +108,26 @@ fn kill_child_tree(child: &tokio::process::Child) {
 
 #[cfg(unix)]
 async fn interrupt_and_wait(child: &mut tokio::process::Child) {
+    use log::error;
     use nix::sys::signal::{killpg, Signal};
     use nix::unistd::{getpgid, Pid};
     use tokio::time::sleep;
 
     if let Some(pid) = child.id() {
-        let gid = getpgid(Some(Pid::from_raw(pid as i32))).unwrap();
-        killpg(gid, Signal::SIGKILL).unwrap();
+        let gid = match getpgid(Some(Pid::from_raw(pid as i32))) {
+            Ok(gid) => gid,
+            Err(error) => {
+                error!(
+                    "Failed to retrieve process group ID of process {pid}, 
+                     cannot proceed with interruption. Error message:\n{error:#?}"
+                );
+                return;
```

Comment:
Is the early return here really correct? Killing the tree seems more consistent

Reply:
Done